kv/memberlist/tcp_transport.go

```diff
@@ -424,27 +449,34 @@ func (t *TCPTransport) getAdvertisedAddr() string {
 // WriteTo is a packet-oriented interface that fires off the given
 // payload to the given address.
 func (t *TCPTransport) WriteTo(b []byte, addr string) (time.Time, error) {
-	t.sentPackets.Inc()
-	t.sentPacketsBytes.Add(float64(len(b)))
+	t.shutdownMu.RLock()
+	defer t.shutdownMu.RUnlock() // Unlock at the end to protect the chan
+	if t.shutdown {
+		return time.Time{}, errors.New("transport is shutting down")
+	}
+	t.writeCh <- writeRequest{b: b, addr: addr}
```

Comment:
If this blocks, it will also block shutdown.

I'd suggest to have some very short "handover timeout" (250ms?), and if it's reached first, silently drop the packet (while increasing some metric), to make sure we don't block here for too long.

WDYT?